core/domain/skill_services.py

```diff
@@ -302,23 +302,12 @@ def get_filtered_skill_summaries(
     """
     augmented_skill_summaries = []
 
-    augmented_skill_summaries_batch, new_urlsafe_start_cursor, more = (
-        _get_augmented_skill_summaries_in_batches(
-            num_skills_to_fetch, urlsafe_start_cursor, sort_by))
-    filtered_augmented_skill_summaries = _filter_skills_by_status(
-        augmented_skill_summaries_batch, status)
-    filtered_augmented_skill_summaries = _filter_skills_by_classroom(
-        filtered_augmented_skill_summaries, classroom_name)
-
-    filtered_augmented_skill_summaries = _filter_skills_by_keywords(
-        filtered_augmented_skill_summaries, keywords)
-    augmented_skill_summaries.extend(filtered_augmented_skill_summaries)
+    more = True
 
-    while (len(augmented_skill_summaries) < num_skills_to_fetch
-           and more and new_urlsafe_start_cursor):
+    while len(augmented_skill_summaries) < num_skills_to_fetch and more:
         augmented_skill_summaries_batch, new_urlsafe_start_cursor, more = (
             _get_augmented_skill_summaries_in_batches(
-                num_skills_to_fetch, new_urlsafe_start_cursor, sort_by))
+                num_skills_to_fetch, urlsafe_start_cursor, sort_by))
```

Comment:
There's a logical error here because you're using the same urlsafe_start_cursor in every iteration. Please write a test that causes it to fail, then fix this code so that the test passes. You should write the failing test first before attempting a fix.

Reply:
It was going in an infinite loop, I have added a test which verifies that if the cursor is present at some point, after another call it should get false(I think this test will suffice.)